### PR TITLE
fuzzing core/state package without skip slot cache

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #4882

---

# Description
Fuzz test state package and fix errors that have been caught
will include skip slot cache in another pr

### CODE REVIEW

File: beacon-chain/core/state/state.go [hunk 1]
```diff
@@ -65,6 +65,9 @@ func GenesisBeaconState(deposits []*ethpb.Deposit, genesisTime uint64, eth1Data
 	// Process initial deposits.
 	leaves := [][]byte{}
 	for _, deposit := range deposits {
+		if deposit == nil || deposit.Data == nil {
+			return nil, errors.New("eth1data contains nil or deposits with nil data field")
```
Comment (terencechain) [inline beacon-chain/core/state/state.go]: shouldn't this just be deposits? where is eth1data?

**Comment by rauljordan** [comment]
PTAL @terencechain

**Comment by shayzluf** [comment]
thanks @rauljordan

File: beacon-chain/core/state/transition.go [hunk 2]
```diff
@@ -563,7 +569,9 @@ func verifyOperationLengths(state *stateTrie.BeaconState, body *ethpb.BeaconBloc
 			params.BeaconConfig().MaxVoluntaryExits,
 		)
 	}
-
+	if state.Eth1Data() == nil {
```
Comment (prestonvanloon) [inline beacon-chain/core/state/transition.go]: This makes a copy of the field accessed, can you assign the result and reuse it on the next if statement so we aren't making multiple copies of the data?

File: beacon-chain/core/state/transition_fuzz_test.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+	ctx := context.Background()
+	state := &stateTrie.BeaconState{}
+	sb := &ethpb.SignedBeaconBlock{}
+	fuzzer := fuzz.NewWithSeed(0)
+	fuzzer.NilChance(0.1)
+	for i := 0; i < 1000; i++ {
+		fuzzer.Fuzz(state)
+		fuzzer.Fuzz(sb)
+		s, err := ExecuteStateTransition(ctx, state, sb)
+		if err != nil {
```
Comment (prestonvanloon) [inline beacon-chain/core/state/transition_fuzz_test.go]: Please simplify to one if block.
```suggestion
		if err != nil && s != nil {
```

Here and everywhere in this file.

File: beacon-chain/core/state/transition.go (hunk 2 above)
Comment (shayzluf) [reply beacon-chain/core/state/transition.go]: thanks!
done

File: beacon-chain/core/state/transition_fuzz_test.go (hunk 3 above)
Comment (shayzluf) [reply beacon-chain/core/state/transition_fuzz_test.go]: thanks . simplified throughout the test file

**Comment by shayzluf** [comment]
@prestonvanloon PTAL

File: beacon-chain/core/state/state_fuzz_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	}
+}
+
+func TestIsValidGenesisState_100000(t *testing.T) {
+	fuzzer := fuzz.NewWithSeed(0)
+	fuzzer.NilChance(0.1)
+	var chainStartDepositCount, currentTime uint64
+	for i := 0; i < 100000; i++ {
+		fuzzer.Fuzz(&chainStartDepositCount)
+		fuzzer.Fuzz(&chainStartDepositCount)
```
Comment (prestonvanloon) [inline beacon-chain/core/state/state_fuzz_test.go]: Why twice?

Comment (shayzluf) [reply beacon-chain/core/state/state_fuzz_test.go]: by mistake 
```suggestion
			fuzzer.Fuzz(&currentTime)
```

**Pull request merged**